bin/fin

```diff
@@ -5127,37 +5203,31 @@ addon_install ()
 	fi
 
 	if [ -d "$addon_path" ] || [ -f "$addon_path" ]; then
-		echo-yellow "Addon already exists in $addon_path"
-		#_confirm "Overwrite?"
-	fi
-	mkdir -p "$addon_path"
-	if_failed_error "Could not create $addon_path" "Check file permissions and try again"
-	local URL_ADDON="$URL_ADDONS_REPO/master/$addon_name"
-
-	# Get files list
-	local file_list
-	file_list=$(curl -fsL "$URL_ADDON/$addon_name.filelist")
-
-	# Download only hook files first
-	local hooks_regex="$addon_name.pre-install\|$addon_name.post-install\|$addon_name.pre-uninstall\|$addon_name.post-uninstall"
-	if [[ "$file_list" != "" ]]; then
-		# Skip line comments, comments are lines that start with '#'
-		file_list=$(echo "$file_list" | grep -v '^#')
-		# Get only hooks
-		local hooks_file_list
-		hooks_file_list=$(echo "$file_list" | grep "$hooks_regex")
-		if [[ "$hooks_file_list" != "" ]]; then
-			echo-green "Downloading addon hook files..."
-			(
-				cd "$addon_path" || exit 1
-				for f in ${hooks_file_list}; do
-					echo "  $f"
-					# Exit subshell with error if download failed
-					curl -fsSL "$URL_ADDON/$f?$RANDOM" -o "$f" || exit 1
-				done
-			)
-			if_failed_error "Download has failed" "Check log above for messages"
-		fi
+		echo-red "Addon already exists in $addon_path. Run ${NC}fin addon update ${addon_name}"
+		exit 1
+	else
+		mkdir -p "$addon_path"
+		if_failed_error "Could not create $addon_path" "Check file permissions and try again"
+	fi
+
+	local tmp_addon="/tmp/${ADDON_UUID}"
+
+	# If repo contains only 1 addon let's download the full repo.
+	if [[ -f $tmp_addon/$addon_name ]] && [[ -x $tmp_addon/$addon_name ]]; then
+		cd $tmp_addon/
+		git pull --depth=1 origin/${URL_ADDONS_BRANCH}
+		# Remove Git Repo that was cloned from.
+		rm -rf $tmp_addon/.git 2>/dev/null
+		mv -R $tmp_addon $addon_path
```

Comment:
`mv -R` is an anti-pattern because of inability to properly handle errors (what failed, copy or delete?) and because it will leave leftovers if copying errors happen.
`cp -R` followed by `rm -rf` should be used instead.